crates/ethereum-instance/src/lib.rs

```diff
@@ -115,19 +109,39 @@ impl EthereumSandbox {
 
         Self {
             _anvil: Some(anvil),
-            client: Arc::new(client),
+            signer: Arc::new(client),
         }
     }
 
     /// Returns local client configured for the running Anvil instance
+    pub fn signer(&self) -> Arc<LocalWalletSignerMiddleware> {
+        self.signer.clone()
+    }
+}
+
+
+/// Wrapper For Spawning a ethereum instance using the provider
+/// Used for deploying on the main network purposes
+pub struct EthereumInstance {
```

Comment:
I kept is separate for connecting the custom signers as ethereum client has properties like anvil and sandbox that may look confusing to the user. If you have any refactor suggestion please do tell will update it.

Reply:
I think we should refactor the attach endpoint of EthereumClient instead so that it works like the attach method of `EthereumInstance` currently. And then we can only use `EthereumClient` and there are two ways to create one - spawn or attach an existing instance. The attach method is agnostic of ANVIL or actual Ethereum.

Reply:
fixed